android/app/src/main/python/mmrelay/plugins/map_plugin.py

```
@@ -0,0 +1,353 @@
+        lat (float): Original latitude
+        lon (float): Original longitude
+        radius (int): Maximum offset distance in meters (default: 1000)
+
+    Returns:
+        tuple: (new_lat, new_lon) with random offset applied
+
+    Adds random offset within specified radius to obscure exact locations
+    while maintaining general geographic area for mapping purposes.
+    """
+    # Generate random offsets for latitude and longitude
+    lat_offset = random.uniform(-radius / 111320, radius / 111320)
```

Ruff (0.12.2)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)